dhis-2/dhis-support/dhis-support-db-migration/src/main/resources/org/hisp/dhis/db/migration/2.42/V2_42_28__Make_ou_event_and_enrollment_not_null.sql

```diff
@@ -0,0 +1,18 @@
+DO $$
+BEGIN
+    -- Set null organisation unit of dummy enrollments to root organisation unit
+    update enrollment en
+    set organisationunitid = (select organisationunitid from organisationunit order by hierarchylevel limit 1)
+    where en.organisationunitid is null
+      and en.programid in (select programid from program where type = 'WITHOUT_REGISTRATION');
+
+    alter table if exists enrollment alter column organisationunitid set not null;
+
+    alter table if exists event alter column organisationunitid set not null;
+
+EXCEPTION
+  WHEN not_null_violation THEN
+    RAISE EXCEPTION 'There are inconsistent data in your DB. Please check https://github.com/dhis2/dhis2-releases/blob/master/releases/2.42/migration-notes.md#null-organisation-unit to have more information on the issue and to find ways to fix it. Detailed error message: %', SQLERRM;
```

Comment:
```suggestion
    RAISE EXCEPTION 'There is inconsistent data in your DB. Please check https://github.com/dhis2/dhis2-releases/blob/master/releases/2.42/migration-notes.md#null-organisation-unit to have more information on the issue and to find ways to fix it. Detailed error message: %', SQLERRM;
```

The URL better not change 😅 maybe a permalink is better, downside is its going to point to outdated docs if you do change its content.